src/core.jl

```diff
@@ -109,6 +118,66 @@ struct Polygon <: Drawable
     vertices::Vector{Point}
 end
 
+"""
+    BoundaryFill <: AbstractPolyFillAlgorithm
+    BoundaryFill(x::Int, y::Int, fill_value::T, boundary_value::T)
+
+    draw(img, verts, alg::BoundaryFill; closed)
+    draw!(img, verts, alg::BoundaryFill; closed)
+
+Applies boundary filling algortithm on image inside the vertices provided in order
+
+# Output
+
+Return the boundary filled image inside the vertices(provided in order). 
+
+# Arguments
+
+## `x` && `y`
+
+Boundary fill is a seeded polygon filling algorithm. 
+So we need to provide the seed point (x,y) inside the image from where the algorithm can start its function.
+// To-do checking if verts and seed are inside the image or note
```

Comment:
Does this mean that you still need extra test cases to check this?

Reply:
I am not sure if it's required, because vertices and seed are supposed to be inside to make logical sense. Since we want to draw inside the image and that seed need to be inside the closed polygon formed when vertices(these are to be provided in order) are given..

Reply:
> I am not sure if it's required, because vertices and seed are supposed to be inside to make logical sense

The `boundscheck` looks good to me. As a regression test, it can be useful to detect if our logic/implementation is correct.

Reply:
Would be fine if writing such a test being too difficult, since we only have very limited image morphology support. (Making it hard to detect the region).